addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -834,29 +831,36 @@ def unpickable_item(self,
                 _('Cannot find move lines todo for unpickable item '
                   'in this batch.'))
 
-        # at this point we should have only one picking_id
-        picking = move_lines.mapped('picking_id')
-        picking.message_post(body=msg)
+        pickings = move_lines.mapped('picking_id')
+        if raise_stock_investigation:
+            to_investigate = Picking.browse()
+        for picking in pickings:
+            picking.message_post(body=msg)
 
-        if picking.batch_id != self:
-            raise ValidationError(_('Move line is not part of the batch.'))
+            if picking.batch_id != self:
+                raise ValidationError(_('Move line is not part of the batch.'))
 
-        if picking.state in ['cancel', 'done']:
-            raise ValidationError(_('Cannot mark a move line as unpickable '
-                                    'when it is part of a completed Picking.'))
+            if picking.state in ['cancel', 'done']:
+                raise ValidationError(_('Cannot mark a move line as unpickable '
+                                        'when it is part of a completed Picking.'))
 
-        original_picking_id = None
+            original_picking_id = None
 
-        if len(picking.move_line_ids - move_lines):
-            # Create a backorder for the affected move lines if
-            # there are move lines that are not affected
-            original_picking_id = picking.id
-            picking = picking._backorder_movelines(move_lines)
+            if len(picking.move_line_ids - move_lines):
+                # Create a backorder for the affected move lines if
+                # there are move lines that are not affected
+                original_picking_id = picking.id
+                picking = picking._backorder_movelines(move_lines)
 
-        moves = picking.move_lines
-        if raise_stock_investigation:
+            if raise_stock_investigation:
+                to_investigate |= picking
+
+            else:
+                picking.batch_id = False
+
+        if raise_stock_investigation and to_investigate:
```

Comment:
Is there any case where we can have `raise_stock_investigation` and not `to_investigate` ?

Reply:
There isn't, but `Picking.raise_stock_inv` gets upset if it's called on an empty recordset.

Reply:
Which I think it should never happen, since to_investigate comes from pickings which comes from move_lines for which we raise an error if it is empty.